### PR TITLE
Re-enable -Wswitch and -Wswitch-enum

### PR DESCRIPTION
**Pull request by aprocter**
These warnings are currently enabled locally in a number of places via pragmas, but I think it'd be a good idea to enable them globally. Doing so actually exposed a couple of small problems, and doesn't lead to as much as you might expect in the way of boiler-plate case enumeration.

### CODE REVIEW

File: src/ngraph/op/max.cpp [hunk 1]
```diff
@@ -52,6 +52,8 @@ shared_ptr<Node> op::Max::get_default_value() const
         return make_constant_from_string("0", get_element_type(), get_shape());
     case element::Type_t::bf16:
         return make_constant_from_string("-INFINITY", get_element_type(), get_shape());
+    case element::Type_t::f16:
+        return make_constant_from_string("-INFINITY", get_element_type(), get_shape());
```
Comment (dcaballe) [inline src/ngraph/op/max.cpp]: nit: looks like all these bf/f cases can be simplified to fall into a single `return make_constant_from_string("-INFINITY", get_element_type(), get_shape());` statement?

File: src/ngraph/op/min.cpp [hunk 2]
```diff
@@ -52,6 +52,8 @@ shared_ptr<Node> op::Min::get_default_value() const
         return make_constant_from_string("1", get_element_type(), get_shape());
     case element::Type_t::bf16:
         return make_constant_from_string("INFINITY", get_element_type(), get_shape());
+    case element::Type_t::f16:
+        return make_constant_from_string("INFINITY", get_element_type(), get_shape());
```
Comment (dcaballe) [inline src/ngraph/op/min.cpp]: Same here

File: src/ngraph/op/max.cpp (hunk 1 above)
Comment (aprocter) [reply src/ngraph/op/max.cpp]: Fixéd.

File: src/ngraph/op/min.cpp (hunk 2 above)
Comment (aprocter) [reply src/ngraph/op/min.cpp]: Fixëd.

**Comment by silee2** [comment]
I see two compile errors. One will be solved by #3229 and the other would need some change in CPU emitter.

**Pull request merged**